File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) { // 분수의 덧셈
    let answer = []
    let numer = (numer1 * denom2) + (numer2 * denom1);
    let denom =  denom1 * denom2;
    const gcd = (numer, denom) => (numer % denom === 0 ? denom : gcd(denom, numer % denom))

    answer[0] = numer / gcd(numer, denom);
    answer[1] = denom / gcd(numer, denom);

    return answer;
}

console.log(solution(1,2,3,4));
console.log(solution(9,2,1,3));

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.go
func solution(numer1, denom1, numer2, denom2 int) []int { // 분수의 덧셈
	numer := (numer1 * denom2) + (numer2 * denom1)
	denom := denom1 * denom2

	// 최대공약수 함수
	gcd := func(a, b int) int {
		for b != 0 {
			a, b = b, a%b
		}
		return a
	}

	g := gcd(numer, denom)

	answer := []int{numer / g, denom / g}

	return answer
}